File: utils/testData.ts
/**
 * Test data factories for BSS Account Payable
 */

/** Login credentials interface */
export interface LoginCredentials {
  username: string;
  password: string;
  subscriberId: string;
}

/**
 * Default valid login credentials (override via env variables)
 */
export function getValidCredentials(): LoginCredentials {
  return {
    username: process.env.USERNAME || 'testuser',
    password: process.env.PASSWORD || 'TestPass123!',
    subscriberId: process.env.SUBSCRIBER_ID || 'BSSQA',
  };
}

/**
 * Invalid login credentials for negative tests
 */
export function getInvalidCredentials(): LoginCredentials {
  return {
    username: 'invaliduser',
    password: 'WrongPassword!',
    subscriberId: 'INVALID',
  };
}

File: java/src/test/java/com/bss/accountpayable/utils/TestData.java
package com.bss.accountpayable.utils;

public final class TestData {
  private TestData() {}

  public record LoginCredentials(String username, String password, String subscriberId) {}

  public static LoginCredentials getValidCredentials() {
    return new LoginCredentials(
        getOrDefault("USERNAME", "testuser"),
        getOrDefault("PASSWORD", "TestPass123!"),
        getOrDefault("SUBSCRIBER_ID", "BSSQA"));
  }

  public static LoginCredentials getInvalidCredentials() {
    return new LoginCredentials("invaliduser", "WrongPassword!", "INVALID");
  }

  private static String getOrDefault(String key, String fallback) {
    String value = System.getenv(key);
    return value == null ? fallback : value;
  }
}
